databuilder/databuilder/extractor/query/json_query_metadata_extractor.py

```diff
@@ -0,0 +1,96 @@
+        [
+            {
+                'sql': "select field1, field2 from table1 where dt = '2021-05-01'",
+                'clean_sql': 'select field1, field2 from table1',   -- Optional
+                'user': 'my_user@my_domain.com',
+                'query_tables': [                                   -- Optional object
+                    {
+                        'database': 'snowflake',                    -- Optional
+                        'cluster': 'default',                       -- Optional
+                        'schema': 'public',                         -- Optional
+                        'table': 'users',
+                    }
+                ],
+                'query_wheres': [                                   -- Optional object
+                    {
+                        'tables': [
+                            {
+                                'database': 'snowflake',            -- Optional
+                                'cluster': 'default',               -- Optional
+                                'schema': 'public',                 -- Optional
+                                'table': 'users',
+                                'columns': ['field']
+                            }
+                        ],
+                        'where_clause': 'where users.field > 3',
+                        'left_arg': 'users.field',                  -- Optional
+                        'right_arg': '3',                           -- Optional
+                        'operator': '>'                             -- Optional
+                    }
+                ],
+                'query_joins': [                                    -- Optional object
+                    {
+                        'left_table': {
+                            'database': 'snowflake',                -- Optional
+                            'cluster': 'default',                   -- Optional
+                            'schema': 'public',                     -- Optional
+                            'table': 'users',
+                            'column': 'user_id'
+                        },
+                        'right_table': {
+                            'database': 'snowflake',                -- Optional
+                            'cluster': 'default',                   -- Optional
+                            'schema': 'public',                     -- Optional
+                            'table': 'events',
+                            'column': 'user_id'
+                        },
+                        'join_type': 'inner join',
+                        'join_operator': '=',
+                        'join_sql': '3'
+                    }
+                ]
+            }
+        ]
+    Implementers should make sure that only VALID and SUCCESSFULLY run queries
+    are included in this list.
+    """
+
+    def __init__(self, query_records: List[Dict]) -> None:
```

Comment:
why we need the constructor here instead of passing it through config?

Reply:
AFAIK, the pyhocon package doesn't support passing dictionaries through it, it converts all of them into Config objects which turn turn loses many of the native python dict functionality. This made it super difficult to then use those dictionaries later on in the processing.

I _could_ pass in a file location through the config and then have the Class read in the contents of the file, but that would be an unnecessary step in the process for many use-cases that are doing query parsing (IMO). Actually spent a bit of time trying to get this to work w/ the config constructor and I'm open to other suggestions if there is a better way.